code/model/modelanimation.h

```diff
@@ -67,10 +66,44 @@ class optional {
 class ship;
 class ship_info;
 
+enum class AnimationTriggerType : int {
+	None = -1,       // No animation
+	Initial,		 // This is just the position the subobject should be placed in
+	Docking_Stage1,	 // Before you dock
+	Docking_Stage2,	 // Before you dock
+	Docking_Stage3,	 // Before you dock
+	Docked,			 // As you dock
```

Comment:
These should probably be annotated based on the modes in ai_dock().  Each animation is played as the docking stage AIS_ begins.

Docking_Stage1 = AIS_DOCK_1 = follow the path until just before the end
Docking_Stage2 = AIS_DOCK_2 = drag oneself right to the second last point on the path
Docking_Stage3 = AIS_DOCK_3 = move directly to the dockpoint using thrusters
Docked = AIS_DOCK_4A or AIS_DOCK_4 = the attachment sound is played and the mission log indicates docking

The old "docking" type was too general and had to be approximated to Docking_Stage2.

Reply:
This is good, but I think the AIS_ macros should be included in the comments as well, as a useful cross-reference